### DISCUSSION TITLE
ASN.1: Support for REAL (floating point) values

### DISCUSSION THREAD

**Discussion by jorgehog** [bcgit/bc-java#1589] [Ideas]
Hi, I don't know if this is the right place to ask, but here goes:
At the company I work we use BC for encoding/decoding asn.1, however, we're at a bit of a pickle because we're going to need real values at some point, which currently isn't parsed (throws exception in the asn.1 parser). 

Is this something BC would ever consider adding support for, or accept a pull request for?

Cheers and thanks in advance :)

**Comment by peterdettman** [comment]
We would accept a PR, but are unlikely to implement it ourselves any time soon.

**Comment by jorgehog** [comment]
A brief status update: 

Support for encoding/decoding Double seem to be working (still need a few more unit tests, comments etc.). 

Extension to Float should be trivial, however, as java isn't my primary language I struggle to find a good way to do this without creating excessive bloat and basically copy pasting all the Double code. So I am a bit lost here. 

Handling arbitrary precision via BigDecimal should also be relatively painless by using base 10 for encoding (basically ascii). 
Decoding from a base != 10 is the pressure point here, but using some algorithm to convert from base 2 to base 10 is likely the way to go.

For my personal use I don't need anything more than Double support. 
And I suspect this also is the case for a lot of people who use ASN.1 Real today. 

So my question is whether you find it reasonable to accept support for Double as a standalone PR, or whether you expect support for Float and BigDecimal as well for completeness. In the odd case where someone needs a Float they can always convert the Double after decoding.

**Reply by jorgehog** [reply]
Clarification: Full X.690 spec is supported for decoding doubles. In other words, we support base 2, 8, 10, 16 with scaling factors etc. No assumptions are made about the the content of the data bytes.
